File: CollaborationBoard/Scripts/App/message.ts
interface ISerializedMessage {
    text: string;
    sender: string;
    senderName: string;
}

class Message {
    public text: string;
    public senderName: string;
    public sender: string;
    public dateSent: string;

    constructor(text: string, sender: string, senderName: string, dateSent: string) {
        this.text = text;
        this.sender = sender;
        this.senderName = senderName;
        this.dateSent = dateSent;
    }

    public serialize(): ISerializedMessage {
        return {
            text: this.text,
            sender: this.sender,
            senderName: this.senderName
        };
    }

    public static deserialize(serialized: any): Message {
        return new Message(serialized.text, serialized.sender, serialized.senderName, serialized.dateSent);
    }
}

File: CollaborationBoard/Source/Message.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Web;

namespace CollaborationBoard
{
    [DataContract]
    public class Message
    {
        [DataMember(Name = "text")]
        public string Text { get; set; }

        [DataMember(Name = "sender")]
        public string Sender { get; set; }

        [DataMember(Name = "senderName")]
        public string SenderName { get; set; }

        [DataMember(Name = "color")]
        public string Color { get; set; }

        [DataMember(Name="dateSent")]
        public DateTime DateSent { get; set; }
    }
}